src/storage/src/table/mod.rs

```diff
@@ -66,47 +65,6 @@ impl Distribution {
     }
 }
 
-// TODO: GAT-ify this trait or remove this trait
```

Comment:
This trait was only `Impl`ed by batch table. Moved it there to keep batch_table interfaces consistent, to return `StorageTableResult` rather than `StorageTable`.

I'm not sure if there's other use for this trait outside of batch table. IF not perhaps further refactor can be done to just place these functions under `impl StorageTable`.